api: panic on splitting pdf

This is my code:
```
package main

import (
	"fmt"
	"log"

	"github.com/pdfcpu/pdfcpu/pkg/api"
)

func main() {
	err := splitPDF("./test.pdf", "output", 10)
	if err != nil {
		fmt.Println("Error split PDF:", err)
	}
}

func splitPDF(inputFile string, outputPrefix string, pageCountPerFile int) error {
	gotPageCount, err := api.PageCountFile(inputFile)
	if err != nil {
		log.Fatalf("%s: %v\n", "err", err)
	}

	log.Printf("pageCount %d \n", gotPageCount)

	if err := api.SplitFile(inputFile, "./res", pageCountPerFile, nil); err != nil {
		log.Fatalf("err: %v\n", err)
	}

	return nil
}

```

I'm getting panic:

```
2024/01/28 18:23:42 pageCount 89 
panic: interface conversion: types.Object is types.Dict, not types.IndirectRef

goroutine 1 [running]:
github.com/pdfcpu/pdfcpu/pkg/pdfcpu.migrateAnnots({0x8d65b0?, 0xc002ecfbd0?}, {0x94, 0x0}, 0xc00297d020, 0xc002e93920?, 0x58b326?)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/migrate.go:98 +0x5e5
github.com/pdfcpu/pdfcpu/pkg/pdfcpu.migratePageDict(0x821da0?, {0x8d6520?, 0xc00301f890?}, 0x8?, 0x7d79a0?, 0xc002208280?)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/migrate.go:144 +0x139
github.com/pdfcpu/pdfcpu/pkg/pdfcpu.addPages(0xc00297d020, 0xc0021479e0, {0xc000024960, 0xa, 0x7fe6bb7ee108?}, 0x0, {0x2, 0x0}, 0x30?, 0xc002e93be8, ...)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/page.go:68 +0x5c5
github.com/pdfcpu/pdfcpu/pkg/pdfcpu.AddPages(0xc00297d020, 0xc0021479e0, {0xc000024960, 0xa, 0xa}, 0xf2?)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/page.go:115 +0x21c
github.com/pdfcpu/pdfcpu/pkg/pdfcpu.ExtractPages(0x789507?, {0xc000024960, 0xa, 0xa}, 0x46?)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/pdfcpu/extract.go:557 +0x8c
github.com/pdfcpu/pdfcpu/pkg/api.pageSpan(0xe?, 0x3d, 0x46)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/split.go:41 +0x7a
github.com/pdfcpu/pdfcpu/pkg/api.writePageSpan(0x82ee44?, 0x5?, 0x831638?, {0xc0015b3890, 0x12})
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/split.go:73 +0x25
github.com/pdfcpu/pdfcpu/pkg/api.writePageSpans(0xc00297d020, 0xa, {0x82ee44, 0x5}, {0x831638, 0x8})
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/split.go:161 +0xf1
github.com/pdfcpu/pdfcpu/pkg/api.Split({0x8d5960?, 0xc000524000?}, {0x82ee44, 0x5}, {0x831638, 0x8}, 0xa, 0xc002f155c0?)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/split.go:269 +0x97
github.com/pdfcpu/pdfcpu/pkg/api.SplitFile({0x831636, 0xa}, {0x82ee44, 0x5}, 0x40d0e8?, 0x10?)
        /home/kalimit/go/pkg/mod/github.com/pdfcpu/pdfcpu@v0.6.0/pkg/api/split.go:293 +0x208
main.splitPDF({0x831636, 0xa}, {0x4062d9?, 0x60?}, 0x0?)
        /home/kalimit/projects/pdp-splitter/main.go:25 +0xeb
main.main()
        /home/kalimit/projects/pdp-splitter/main.go:11 +0x3a
exit status 2

```

I use Ubuntu 23.04, go1.20.3 linux/amd64

--- a/pkg/pdfcpu/migrate.go
+++ b/pkg/pdfcpu/migrate.go
@@ -95,7 +95,10 @@ func migrateAnnots(o types.Object, pageIndRef types.IndirectRef, ctxSrc, ctxDest
 	}
 
 	for i, v := range arr {
-		o := v.(types.IndirectRef)
+		o, ok := v.(types.IndirectRef)
+		if !ok {
+			continue
+		}
 		objNr := o.ObjectNumber.Value()
 		if migrated[objNr] > 0 {
 			o.ObjectNumber = types.Integer(migrated[objNr])
@@ -107,7 +110,10 @@ func migrateAnnots(o types.Object, pageIndRef types.IndirectRef, ctxSrc, ctxDest
 			return nil, err
 		}
 		arr[i] = o
-		d := o1.(types.Dict)
+		d, ok := o1.(types.Dict)
+		if !ok {
+			continue
+		}
 		for k, v := range d {
 			if k == "P" {
 				d["P"] = pageIndRef
@@ -179,7 +185,10 @@ func migrateFields(d types.Dict, fieldsSrc, fieldsDest *types.Array, ctxSrc, ctx
 			continue
 		}
 		for _, v := range *fieldsSrc {
-			ir := v.(types.IndirectRef)
+			ir, ok := v.(types.IndirectRef)
+			if !ok {
+				continue
+			}
 			objNr := ir.ObjectNumber.Value()
 			if migrated[objNr] == indRef.ObjectNumber.Value() {
 				*fieldsDest = append(*fieldsDest, indRef)
@@ -224,7 +233,10 @@ func migrateFormDict(d types.Dict, fields types.Array, ctxSrc, ctxDest *model.Co
 
 func detectMigratedAnnot(ctxSrc *model.Context, indRef *types.IndirectRef, kids types.Array, migrated map[int]int) (bool, error) {
 	for _, v := range kids {
-		ir := v.(types.IndirectRef)
+		ir, ok := v.(types.IndirectRef)
+		if !ok {
+			continue
+		}
 		objNr := ir.ObjectNumber.Value()
 		if migrated[objNr] == indRef.ObjectNumber.Value() {
 			return true, nil

